libyul/optimiser/Metrics.cpp

```diff
@@ -36,30 +36,71 @@ using namespace solidity;
 using namespace solidity::yul;
 using namespace solidity::util;
 
-size_t CodeSize::codeSize(Statement const& _statement)
+size_t CodeWeights::costOf(Statement const& _statement) const
 {
-	CodeSize cs;
+	if (holds_alternative<ExpressionStatement>(_statement))
```

Comment:
Would it be better to use `GenericVisitor` here?

Reply:
I'm not sure it would improve anything when it comes to readability. The `if` is pretty succint. I could even put each branch on one line to make it even clearer.

Personally I prefer the `if` because it's dead simple, but I can change it if you still want `GenericVisitor` here.

Reply:
Yeah, I think it does not make a big difference.